Tests/Research/QuantBookFundamentalTests.cs

```diff
@@ -0,0 +1,167 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using NUnit.Framework;
+using Python.Runtime;
+using System;
+using System.Collections.Generic;
+using QuantConnect.Data.Market;
+using QuantConnect.Research;
+using QuantConnect.Securities;
+
+namespace QuantConnect.Tests.Research
+{
+    [TestFixture]
+    public class QuantBookFundamentalTests
+    {
+        private dynamic _module;
+        private DateTime _startDate;
+        private DateTime _endDate;
+
+        [SetUp]
+        public void Setup()
+        {
+            SymbolCache.Clear();
+            MarketHoursDatabase.Reset();
+
+            // Using a date that we have data for in the repo
+            _startDate = new DateTime(2014, 3, 31);
+            _endDate = new DateTime(2014, 4, 1);
+
+            using (Py.GIL())
+            {
+                _module = Py.Import("Test_QuantBookFundamental");
+            }
+        }
+
+        // All 4 of the accepted types of input for fundamental data request, with varying selectors
+        private static readonly object[] FundamentalRequestTestCases =
```

Comment:
nit - code style is to place privates at the bottom. First setup methods, second all tests and finally any private helper classes/methods